### PR TITLE
Fix: relax MTP logits validation tolerance

### PR DESCRIPTION
**Pull request by Little-oil**
## Problem

DeepSeek V4 Flash MTP intermittently fails only the `logits` precision check, while `kv_cache`, `hidden_out`, and `pre_hc_hidden_out` all pass. In two previous failures, the proportions of logits exceeding the per-element error threshold were:

- 71,339 / 2,068,480 = 3.4489%
- 90,765 / 2,068,480 = 4.3880%

The original check allowed only 1% of elements to exceed the threshold, causing intermittent failures.

## Changes

This change keeps the existing CPU golden outputs and direct logits comparison, and only increases `max_error_ratio` from 1% to 5%. The net diff of this PR is a single line.

Why 5%:

- 5% covers the historical worst case of 4.3880% while retaining a margin of approximately 0.612 percentage points.
- It matches the 5% ratio already used for `hidden_out` and `pre_hc_hidden_out` in the same MTP test; logits can amplify the allowed error in upstream hidden states.
- The per-element criteria remain unchanged at `atol=1e-2` and `rtol=1e-2`, so the error tolerance for individual elements has not been relaxed.
- `ratio_allclose` still fails immediately if the actual output contains NaN or Inf values.
- 5% remains stricter than the 10% used for decode MTP logits.

## Validation

- `pytest -q tests/golden`: 169 passed
- Ruff, py_compile, and `git diff --check`: passed
- Current toolchain, even-numbered devices 0 and 2: `task_20260803_180715_349271717343`
  - `kv_cache`, `hidden_out`, `pre_hc_hidden_out`, and `logits` all passed
  - Logs confirm that `ratio_allclose(atol=0.01, rtol=0.01, max_error_ratio=0.05)` was used
- [CI run 30867846363](https://github.com/hw-native-sys/pypto-lib/actions/runs/30867846363): all applicable checks passed, including a2a3, a2a3sim, a5sim, and serving-deepseek.

### CODE REVIEW

File: models/deepseek/v4-flash/lm_head_validation.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        lm_head_ok, lm_head_detail = base_cmp(
+            actual,
+            reference,
+            actual_outputs=actual_outputs,
+            expected_outputs=expected_outputs,
+            inputs=inputs,
+            rtol=rtol,
+            atol=atol,
+        )
+        if not lm_head_ok:
```
[the remark this answers is not in the document]
Comment (Little-oil) [reply models/deepseek/v4-flash/lm_head_validation.py]: Fixed in 8ba2bd6: the comparator now rejects a non-finite projected reference before ratio_allclose. Added regression coverage for both NaN and Inf; the focused tests pass 11/11, all golden tests pass 180/180, and Ruff passes.

**Comment by Little-oil** [comment]
CI retry triage for empty commit `437b3ed`:

- [Run 30806297190](https://github.com/hw-native-sys/pypto-lib/actions/runs/30806297190) did **not** reproduce the logits precision mismatch.
- `a2a3` job `91665122786`: `prefill_mtp.py` reached runtime, then failed with `S1:running-stalled` / AICPU `507018` / runtime `-100`; tensor copy-back and all output comparisons were skipped.
- `a2a3sim` job `91665122784` and `a5sim` job `91665122715`: the preceding examples passed, then both stalled in unchanged `decode_mtp.py` runtime and hit the 30-minute job limit. Neither reached precision validation.
- `unit-tests`, `pre-commit`, `detect-changes`, and `serving-deepseek` passed.

The first attempt also passed the hardware `a2a3` and `serving-deepseek` jobs, and the current upstream base reproduces the simulator runtime stall locally. The remaining red checks are runtime/simulator stalls rather than an output-validation regression.

**Comment by Little-oil** [comment]
Reworked in `63fbcc4` to the requested minimal approach:

- removed the device-hidden reference recomputation, custom comparator, helper tests, and seed changes
- restored the original CPU golden/direct logits comparison
- net diff is one line: logits `max_error_ratio` 1% -> 5%
- local even-card task `task_20260803_180715_349271717343`: all four MTP outputs PASS
- [CI run 30867846363](https://github.com/hw-native-sys/pypto-lib/actions/runs/30867846363): all applicable checks PASS, including a2a3, a2a3sim, a5sim, and serving-deepseek
